src/shared_definitions.hpp

```diff
@@ -194,17 +195,26 @@ struct fake_index_file
     }
 };
 
-template <typename TTargetAlph, typename TRange, typename TAdapt>
+template <typename TTargetAlph, typename TRange, typename TAdaptProt, typename TAdaptNucl>
     requires std::Same<seqan3::innermost_value_type_t<TRange>, TTargetAlph>
-TRange & initHelper(TRange & input, TAdapt &&)
+TRange & initHelper(TRange & input, TAdaptProt &&, TAdaptNucl &&)
 {
     return input;
 }
 
-template <typename TTargetAlph, typename TRange, typename TAdapt>
-auto initHelper(TRange & input, TAdapt && adapt)
+template <typename TTargetAlph, typename TRange, typename TAdaptProt, typename TAdaptNucl>
+    requires seqan3::NucleotideAlphabet<TTargetAlph> &&
+             !std::Same<seqan3::innermost_value_type_t<TRange>, TTargetAlph>
```

Comment:
```suggestion
             (!std::Same<seqan3::innermost_value_type_t<TRange>, TTargetAlph>)
```
As of GCC10 (and with Clang later) we need to put parens around requires expressions that are not "primary expressions" (i.e. everything that contains operators other than `&&` and `||`).